addons/sourcemod/scripting/shop.sp

```diff
@@ -680,7 +680,10 @@ public int OnItemSelect(Menu menu, MenuAction action, int param1, int param2)
 			char info[16];
 			menu.GetItem(param2, info, sizeof(info));
 			iPos[param1] = GetMenuSelectionPosition();
-			if (!ShowItemInfo(param1, StringToInt(info)))
+
+			Action result = Forward_OnItemSelect(param1, bInv[param1] ? Menu_Inventory : Menu_Buy, iClCategoryId[param1], StringToInt(info));
+			
+			if (result == Plugin_Handled || (result == Plugin_Changed || result == Plugin_Continue && !ShowItemInfo(param1, StringToInt(info))))
```

Comment:
а если Plugin_Stop ?
в условии с || и && лучше указывать скобки чтобы явно видеть приоритеты операций
я бы вынес в метод чтобы не городить кучу условий в одном ифе и тогда StringToInt(info) будет только 1 раз выполняться

Reply:
В описании возвратов форварда всё описано.
`Plugin_Handled to block an item to select, and show items of category. Plugin_Stop to close menu. Plugin_Continue or Plugin_Changed otherwise`

Reply:
у  && приоритет выше чем у  || по крайней мере в си
проверь будет ли отображаться итем если вернуть Plugin_Changed,  мне кажется нет
https://www.viva64.com/ru/t/0064/

Reply:
> у && приоритет выше чем у || по крайней мере в си
> проверь будет ли отображаться итем если вернуть Plugin_Changed, мне кажется нет
> https://www.viva64.com/ru/t/0064/

Если сделать так, то всё работает, приоритет разбит дужками (это пофикшеный)
`if (result == Plugin_Handled || ((result == Plugin_Changed || result == Plugin_Continue) && !ShowItemInfo(param1, StringToInt(info))))`

Если не нравится, можно сделать так:
```
Action result = Forward_OnItemSelect(param1, bInv[param1] ? Menu_Inventory : Menu_Buy, iClCategoryId[param1], StringToInt(info));
			
bool nodrawItem = false;

if(result == Plugin_Handled)
	nodrawItem = true;
if(!ShowItemInfo(param1, StringToInt(info)) && (result == Plugin_Changed || result == Plugin_Continue))
	nodrawItem = true;

if (nodrawItem)
{
	ShowItemsOfCategory(param1, iClCategoryId[param1], bInv[param1], iPos[param1]);
}
```

Reply:
ну я уже и на дужки согласен. но хотя бы переносы правильно поставь. чтобы смотрелось логично и просто читалось

понял в чем ошибка была?)